state-synchronizer/src/state_synchronizer.rs

```diff
@@ -120,8 +178,10 @@ impl StateSynchronizer {
         }
     }
 
-    pub fn create_client(&self) -> Arc<StateSyncClient> {
-        Arc::new(StateSyncClient::new(self.coordinator_sender.clone()))
+    pub fn create_client(&self) -> Arc<StateSynchronizerClient> {
```

Comment:
I wonder why we're creating a clone on the client versus just sending out a cloned client. This seems really .. weird. Like why aren't we cloning an arc to begin with... might be worth stripping out the Arc or understanding why it is partially there.

Reply:
Yeah, I thought this was odd too.. I've added a new commit that strips out the Arc because I've yet to see any actual use/reliance on it.. 😕 As far as I can tell, "create_client" is only used in two places:
1. setup_environment in diem node (which only creates a single client)
2. setup_next_synchronizer in integration tests, but there is only a single client per state synchronizer and they aren't shared across threads....

Perhaps this was a legacy thing?